lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -123,10 +123,8 @@ defmodule ExUnit.Assertions do
         end
       end)
 
-    quote do
-      right = unquote(right)
-      expr  = unquote(code)
-      unquote(vars) =
+    match_case =
```

Comment:
Agree, `match_case` just for a seconds looked confusing to me, maybe `match_expr` would be even better?